### PR TITLE
Add OIDC support for the Self-Service UI

### PR DESCRIPTION
**Pull request by abellotti**
Add OIDC Support for the Self-Service UI

### CODE REVIEW

**Comment by abellotti** [comment]
marking as WIP - just taking a while guess on where this goes, and no clue on how to build & test operators.

File: manageiq-operator/pkg/helpers/miq-components/httpd_conf.go [hunk 1]
```diff
@@ -250,6 +250,11 @@ OIDCOAuthIntrospectionEndpointAuth client_secret_post
   Require                    valid-user
 </Location>
 
+<Location /ui/service/oidc_login>
+  AuthType                   openid-connect
+  Require                    valid-user
+  Header set Set-Cookie      "miq_oidc_access_token=%{OIDC_access_token}e; Max-Age=10; Path=/ui/service"
```
Comment (carbonin) [inline manageiq-operator/pkg/helpers/miq-components/httpd_conf.go:256]: I think this `%` needs to be escaped like `%%` as we're using this string in `Sprintf`

**Comment by abellotti** [comment]
Ok, with great help from @carbonin, got this thing tested and verified on OpenShift 4.3.8. (CRC running on mac).

Tested with Keycloak  10.0.2 on OpenShift in the same CRC cluster.

**Self-Service UI Login Screen:**
Screen Shot 2020-06-25 at 5 25 55 PM

**Login Screen From Keycloak:**
Screen Shot 2020-06-25 at 5 26 48 PM

**Self-Service UI:**
Screen Shot 2020-06-25 at 5 13 59 PM

**Keycloak Configuration for that Client in the CRC Cluster:**
Screen Shot 2020-06-25 at 5 16 50 PM

Screen Shot 2020-06-25 at 5 17 08 PM

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit 92cc53207d24e9e61f492582fc309773533509ea
Author: Nick Carboni <ncarboni@redhat.com>
Date:   Fri Jun 26 09:32:45 2020 -0400

    Merge pull request #559 from abellotti/sui-sso

    Add OIDC support for the Self-Service UI

    (cherry picked from commit 6c8c65855c336e0e39328d5b245761837ebff510)
```

**Pull request merged**